src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -79,67 +72,28 @@ class GitVersionPlugin implements Plugin<Project> {
 
     @Memoized
     private Git gitRepo(Project project) {
-        File gitDir = GitCli.getRootGitDir(project.projectDir);
+        File gitDir = GitCli.getRootGitDir(project.projectDir)
         return Git.wrap(new FileRepository(gitDir))
     }
 
     @Memoized
     private String gitDescribe(Project project, String prefix) {
-        // verify that "git" command exists (throws exception if it does not)
-        GitCli.verifyGitCommandExists()
-
-        def runGitCmd = { String... commands ->
-            return GitCli.runGitCommand(project.projectDir, commands);
-        }
-
-        Git git = gitRepo(project)
-        try {
-            // back-compat: the JGit "describe" command throws an exception in repositories with no commits, so call it
-            // first to preserve this behavior in cases where this call would fail but native "git" call does not.
-            new DescribeCommand(git.getRepository()).call()
-
-            /*
-             * Mimick 'git describe --tags --always --first-parent --match=${prefix}*' by using rev-list to
-             * support versions of git < 1.8.4
-             */
-
-            // Get SHAs of all tags, we only need to search for these later on
-            Set<String> tagRefs = Sets.newHashSet()
-            for (String tag : getLines(runGitCmd("show-ref", "--tags", "-d"))) {
-                List<String> parts = WORD_SPLITTER.splitToList(tag)
-                Preconditions.checkArgument(parts.size() == 2, "Could not parse output of `git show-ref`: %s", parts)
-                tagRefs.add(parts.get(0))
+        // TODO(mbakovic): Switch to jgit only implementation
+        String nativeGitDescribe = new NativeGitDescribe(project.projectDir).describe(prefix)
+        String jgitDescribe = new JGitDescribe(project.projectDir).describe(prefix)
+        if (nativeGitDescribe == null) {
+            return jgitDescribe
+        } else if (jgitDescribe == null) {
+            return nativeGitDescribe
+        } else {
+            if (nativeGitDescribe != jgitDescribe) {
+                throw new IllegalStateException(String.format(
+                        "Inconsistent git describe, native was %s and jgit was %s", nativeGitDescribe, jgitDescribe))
```

Comment:
"Please report this on `github.com/palantir/gradle-git-version`"

Reply:
+1